examples/games/game_menu.rs

```diff
@@ -400,380 +465,154 @@ mod menu {
-                                    style: button_icon_style.clone(),
-                                    image: UiImage::new(icon),
-                                    ..default()
-                                });
-                                parent.spawn(TextBundle::from_section(
-                                    "Settings",
-                                    button_text_style.clone(),
-                                ));
-                            });
-                        parent
-                            .spawn((
-                                ButtonBundle {
-                                    style: button_style,
-                                    image: UiImage::default().with_color(NORMAL_BUTTON),
-                                    ..default()
-                                },
-                                MenuButtonAction::Quit,
-                            ))
-                            .with_children(|parent| {
-                                let icon = asset_server.load("textures/Game Icons/exitRight.png");
-                                parent.spawn(ImageBundle {
-                                    style: button_icon_style,
-                                    image: UiImage::new(icon),
-                                    ..default()
-                                });
-                                parent.spawn(TextBundle::from_section("Quit", button_text_style));
-                            });
                     });
-            });
+                    parent.spawn(TextBundle::from_section(text, button_text_style()));
+                });
+            }
+        });
     }
 
     fn settings_menu_setup(mut commands: Commands) {
-        let button_style = Style {
-            width: Val::Px(200.0),
-            height: Val::Px(65.0),
-            margin: UiRect::all(Val::Px(20.0)),
-            justify_content: JustifyContent::Center,
-            align_items: AlignItems::Center,
-            ..default()
-        };
-
-        let button_text_style = TextStyle {
-            font_size: 40.0,
-            color: TEXT_COLOR,
-            ..default()
-        };
+        inner_node(&mut commands, OnSettingsMenuScreen, RED, |parent| {
+            // Display three buttons for each action available from the main settings menu
+            for (action, text) in [
+                (MenuButtonAction::SettingsDisplay, "Display"),
+                (MenuButtonAction::SettingsSound, "Sound"),
+                (MenuButtonAction::BackToMainMenu, "Back"),
+            ] {
+                button(parent, action, |parent| {
+                    parent.spawn(TextBundle::from_section(text, button_text_style()));
+                });
+            }
+        });
+    }
 
-        commands
+    // A button to go back to the main settings menu from settings submenus
+    fn back_to_settings(parent: &mut ChildBuilder) {
+        // Display the back button to return to the settings screen
+        parent
```

Comment:
This can use the `button` helper function.

Reply:
Nice catch. Fixed in next commit.